src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1751,6 +1874,37 @@
       productions
       (throw (ex-info (str "Non-unique production names: " non-unique) {:names non-unique})))))
 
+(def forms-per-eval-default
+  ;; The default max number of forms that will be evaluated together as a single batch.
```

Comment:
I like the analysis. The only thing that is odd to me is you used inline comments instead of making this a doc string of the var. I don't suppose it matters much though.